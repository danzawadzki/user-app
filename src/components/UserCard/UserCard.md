Default UserCard component with all props:

```jsx
<UserCard
	name="Random User"
	location="New York, USA"
	avatar={{
		small: 'https://randomuser.me/api/portraits/thumb/men/65.jpg',
		regular: 'https://randomuser.me/api/portraits/med/men/65.jpg',
		large: 'https://randomuser.me/api/portraits/men/65.jpg'
	}}
	onHeartClick={e => console.log(e.target + ' has been clicked.')}
/>
```
