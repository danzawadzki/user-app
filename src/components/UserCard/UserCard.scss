@import '../../assets/styles/colors';
@import '../../assets/styles/minor';

.UserCard {
	display: flex;
	align-items: center;

	&__avatar {
		padding-right: 2rem;
	}

	&__details {
		& .details {
			&__name {
				font-size: 1.8rem;
				letter-spacing: 0;
				font-weight: 400;
				color: $blue__navy;
				margin: 0;
				display: flex;
				align-items: center;

				&--lighter {
					color: $gray--light;
					font-size: 1.25rem;
					margin: 0 0.75rem;
					transition: $transition--default;
					cursor: pointer;

					&:hover {
						color: $brand--accent;
						transition: $transition--default;
					}
				}
			}

			&__location {
				font-size: 1.2rem;
				color: $blue--light;
				margin: 0;
			}
		}
	}
}
