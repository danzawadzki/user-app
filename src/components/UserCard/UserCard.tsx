import * as React from 'react';
import './UserCard.scss';
import Avatar, { IAvatarSrc } from '../Avatar/Avatar';
import { library } from '@fortawesome/fontawesome-svg-core';
import { faHeart } from '@fortawesome/free-regular-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export interface IUserCard {
	/** User name */
	name: string;
	/** User location */
	location?: string | null;
	/** Object containing avatar source */
	avatar?: IAvatarSrc;
	/** Handler to heart icon onClick action */
	onHeartClick: (e: React.MouseEvent<HTMLSpanElement>) => void;
}

/**
 * UserCard component.
 *
 * @version 1.0.0
 */
const UserCard = ({ name, location = 'Unknown', avatar, onHeartClick }: IUserCard) => {
	/**
	 * Registering font-awesome in the library
	 *
	 * @source https://fontawesome.com/how-to-use/on-the-web/using-with/react
	 */
	library.add(faHeart);

	return (
		<div className="UserCard">
			<div className="UserCard__avatar">
				<Avatar name={name} avatar={avatar ? avatar : null} />
			</div>
			<div className="UserCard__details">
				<div className="details">
					<h1 className="details__name">
						{name}{' '}
						<span className="details__name--lighter" onClick={onHeartClick}>
							<FontAwesomeIcon icon={['far', 'heart']} />
						</span>
					</h1>
					<p className="details__location">{location}</p>
				</div>
			</div>
		</div>
	);
};

export default UserCard;
