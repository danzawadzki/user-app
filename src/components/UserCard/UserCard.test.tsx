import React from 'react';
import { shallow } from 'enzyme';
import UserCard from './UserCard';

describe('UserCard component: ', () => {
	//Shallow rendered component
	const name = 'Random User';
	const location = 'New York, USA';
	const avatar = {
		small: 'https://randomuser.me/api/portraits/thumb/men/65.jpg',
		regular: 'https://randomuser.me/api/portraits/med/men/65.jpg',
		large: 'https://randomuser.me/api/portraits/men/65.jpg'
	};
	const onClick = () => console.log('Heart icon has been clicked.');

	const component = shallow(
		<UserCard
			name={name}
			location={location}
			avatar={avatar}
			onHeartClick={onClick}
		/>
	);

	it('should render match snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	it('should contain user name and location', () => {
		expect(component.html()).toContain(name);
		expect(component.html()).toContain(location);
	});

	it('should render with out location', () => {
		const componentWithOutLocation = shallow(
			<UserCard name={name} avatar={avatar} onHeartClick={onClick} />
		);

		expect(componentWithOutLocation.html()).toContain('Unknown');
	});

	// it('should have length of counters array', () => {
	// 	expect(component.find('li')).toHaveLength(counters.length);
	// });
	//
	// it('should render message "No counters...", if no counters were provided', () => {
	// 	const componentEmpty = shallow(<CountersList />);
	//
	// 	expect(componentEmpty.html()).toContain('No counters...');
	// });
});
