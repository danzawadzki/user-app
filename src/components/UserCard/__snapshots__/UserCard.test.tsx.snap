// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard component:  should render match snapshot 1`] = `
<div
  className="UserCard"
>
  <div
    className="UserCard__avatar"
  >
    <Avatar
      avatar={
        Object {
          "large": "https://randomuser.me/api/portraits/men/65.jpg",
          "regular": "https://randomuser.me/api/portraits/med/men/65.jpg",
          "small": "https://randomuser.me/api/portraits/thumb/men/65.jpg",
        }
      }
      name="Random User"
    />
  </div>
  <div
    className="UserCard__details"
  >
    <div
      className="details"
    >
      <h1
        className="details__name"
      >
        Random User
         
        <span
          className="details__name--lighter"
          onClick={[Function]}
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Array [
                "far",
                "heart",
              ]
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
        </span>
      </h1>
      <p
        className="details__location"
      >
        New York, USA
      </p>
    </div>
  </div>
</div>
`;
